validator/sawtooth_validator/journal/chain.py

```diff
@@ -567,6 +573,9 @@ def on_block_validated(self, commit_new_block, result):
 
                     LOGGER.info("Chain head updated to: %s", self._chain_head)
 
+                    # Publish the state deltas
+                    self._state_delta_processor.publish_deltas(new_block)
```

Comment:
This looks like it could be time consuming, We should do it as the last step in this function. Getting descendant blocks processing and the BlockPublisher building a new block should be done first.